src/main/java/io/kestra/plugin/typesense/BulkIndex.java

```diff
@@ -51,33 +56,76 @@
         )
     }
 )
-public class BulkIndex extends AbstractTypesenseTask implements RunnableTask<VoidOutput> {
+public class BulkIndex extends AbstractTypesenseTask implements RunnableTask<BulkIndex.Output> {
 
     @Schema(
         title = "The file URI containing the documents to index"
     )
+    @NotNull
     private Property<String> from;
 
+    @Schema(
+        title = "The chunk size for every bulk request",
+        description = "Default value is 1000"
+    )
+    private Property<Integer> chunk;
+
     @Override
-    public VoidOutput run(RunContext runContext) throws Exception {
+    public BulkIndex.Output run(RunContext runContext) throws Exception {
         Client client = getClient(runContext);
         String renderedCollection = renderCollection(runContext);
+        Logger logger = runContext.logger();
 
         URI uri = new URI(renderString(from, runContext));
 
         try (
             BufferedReader inputStream = new BufferedReader(
                 new InputStreamReader(runContext.storage().getFile(uri)), FileSerde.BUFFER_SIZE);
         ) {
-            List documents = FileSerde.readAll(inputStream, List.class).blockLast();
-
-            ImportDocumentsParameters queryParameters = new ImportDocumentsParameters();
-            queryParameters.action(IndexAction.UPSERT);
-            client.collections(renderedCollection)
-                .documents()
-                .import_(documents, queryParameters);
+            AtomicLong count = new AtomicLong();
+            Long requestCount = FileSerde.readAll(inputStream)
+                .doOnNext(l -> count.incrementAndGet())
+                .buffer(runContext.render(chunk).as(Integer.class).orElse(1000))
+                .flatMap(documents -> bulkIndex(client, renderedCollection, documents, logger))
+                .count().blockOptional().orElse(0L);
+            runContext.metric(Counter.of("requests.count", requestCount));
+            runContext.metric(Counter.of("records", count.get()));
+            logger.info(
+                "Successfully send {} requests for {} records",
+                requestCount,
+                count.get()
+            );
+            return Output.builder()
+                .size(count.get())
+                .build();
         }
-        return null;
+    }
+
+    private static Flux<String> bulkIndex(Client client, String collection, List documents,
+        Logger logger) {
+        return Flux.defer(() -> {
```

Comment:
I'm not sure Flux.defer() is mandatory here but it didn't hurt neither

Reply:
The flux.defer() helps handling the client exception throw. I can return flux.error(e) this way